src/native/corehost/browserhost/CMakeLists.txt

```diff
@@ -0,0 +1,96 @@
+# Licensed to the .NET Foundation under one or more agreements.
+# The .NET Foundation licenses this file to you under the MIT license.
+
+project(browserhost)
+set(DOTNET_PROJECT_NAME "browserhost")
+
+configure_file(${CLR_SRC_NATIVE_DIR}/corehost/configure.h.in ${GENERATED_INCLUDE_DIR}/corehost/configure.h)
+target_include_directories(hostmisc_interface INTERFACE ${GENERATED_INCLUDE_DIR}/corehost)
+
+set(SOURCES
+    ./browserhost.cpp
+    native.rc
+)
+set(HEADERS
+)
+
+add_compile_definitions(FEATURE_APPHOST)
+add_definitions(-DFEATURE_APPHOST=1)
+add_definitions(-DFEATURE_STATIC_HOST=1)
+
+add_executable(browserhost ${SOURCES})
+set_target_properties(browserhost PROPERTIES OUTPUT_NAME dotnet.native)
+set(CMAKE_EXECUTABLE_SUFFIX ".js")
+
+install(TARGETS browserhost DESTINATION corehost COMPONENT runtime)
+install(FILES ${CMAKE_CURRENT_BINARY_DIR}/dotnet.native.wasm DESTINATION corehost COMPONENT runtime)
+
+include(configure.cmake)
+
+LIST(APPEND NATIVE_LIBS
+    coreclr_static
+    System.Native.Browser-Static
+    System.Runtime.InteropServices.JavaScript.Native-Static
+    System.Native-Static
+    System.Globalization.Native-Static
+    System.IO.Compression.Native-Static
+    nativeresourcestring
+    gcinfo
+    # WASM-TODO respect  $(InvariantTimezone)
+    # System.Native.TimeZoneData.Invariant
+    System.Native.TimeZoneData
+)
+
+# WASM-TODO set(RUNTIMEINFO_LIB runtimeinfo)
+
+target_compile_options(browserhost PRIVATE -fwasm-exceptions)
+
+set_target_properties(browserhost PROPERTIES
+    LINK_DEPENDS "${CLR_ARTIFACTS_OBJ_DIR}/native/browser-${CMAKE_BUILD_TYPE}-wasm/System.Native.Browser/libSystem.Native.Browser.js;${CLR_ARTIFACTS_OBJ_DIR}/native/browser-${CMAKE_BUILD_TYPE}-wasm/System.Runtime.InteropServices.JavaScript.Native/libSystem.Runtime.InteropServices.JavaScript.Native.js;${CLR_ARTIFACTS_OBJ_DIR}/coreclr/browser.wasm.${CMAKE_BUILD_TYPE}/corehost/libBrowserHost.js;${CMAKE_CURRENT_SOURCE_DIR}/../../libs/System.Native.Browser/libSystem.Native.Browser.extpost.js;"
```

Comment:
could we use variables for these huge string literals that are assembled piecemeal? would make it easier to see what's going on, especially because there's some redundancy